### ISSUE TITLE
2.0 - intellijPlatform in dependencies block is not usable from Groovy

### ISSUE THREAD

**Issue by hfhbd** [JetBrains/intellij-platform-gradle-plugin#1592]
### What happened?

In Groovy, the IntelliJPlatform block is not useable, it is in Kotlin.

### Relevant log output or stack trace

```shell
UnsupportedNotationException: Cannot convert the provided notation to an object of type Dependency: build_3gi1eyii8kt0q7ejmxy5fdn8t$_run_closure2@8843b0f.
The following types/formats are supported:
  - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
  - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
  - FileCollections, for example files('some.jar', 'someOther.jar').
  - Projects, for example project(':some:project:path').
  - ClassPathNotation, for example gradleApi().

Comprehensive documentation on dependency notations is available in DSL reference for DependencyHandler type
```

### Steps to reproduce

```groovy
plugins {
  id("org.jetbrains.intellij.platform") version "2.0.0-beta1"
}

dependencies {
  intellijPlatform {
    intellijIdeaCommunity(libs.versions.ideaVersion)
    pluginVerifier("latest")
    javaCompiler(libs.versions.ideaVersion)
  }
}
```

### Gradle IntelliJ Plugin version

2.0.0-beta1

### Gradle version

8.7

### Operating System

macOS

**Comment by YannCebron** [comment]
see also #1591

FTR Kotlin is now the preferred language for Gradle build files

**Comment by hsz** [comment]
This issue was caused by the conflicting IntelliJ Platform configuration and extension names.
Fixed by renaming the configuration from `intellijPlatform` to `intellijPlatformDependency`.

The fix along with initial Groovy support is now available in `2.0.0-SNAPSHOT` release.
To use the snapshot releases, please follow: https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin.html#snapshot-release
